plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/reporter/CustomReporter.kt

```diff
@@ -1,41 +1,26 @@
 package org.jlleitschuh.gradle.ktlint.reporter
 
-import org.gradle.api.artifacts.Dependency
-import org.gradle.api.artifacts.dsl.DependencyHandler
+import java.io.Serializable
 
-class CustomReporter(
+/**
+ * @param name required for Groovy interop, same as [reporterId]
+ * @param reporterId an id that reporter exposes for ktlint `ServiceLocator`
+ * @param fileExtension generated report file extension
+ * @param dependency reporter [dependency notation](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.dsl.DependencyHandler.html#N17198).
+ *
+ * For example, [dependency] could have following notation:
+ * ```
+ * "some.group:reporter:0.1.0"
+ * project(":custom:reporter")
+ * ```
+ */
+data class CustomReporter(
     val name: String,
-    private val dependencyHandler: DependencyHandler
-) {
-    /**
-     * An id that reporter exposes for ktlint `ServiceLocator`.
-     */
-    val reporterId: String = name
-
-    /**
-     * Generated report file extension.
-     */
-    var fileExtension: String = reporterId
-
-    /**
-     * Reporter [dependency notation](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.dsl.DependencyHandler.html#N17198).
-     *
-     * For example it could be:
-     * ```
-     * "some.group:reporter:0.1.0"
-     * project(":custom:reporter")
-     * ```
-     */
-    var dependency: Any? = null
-        set(value) {
-            field = value
-            if (value != null) _dependencyArtifact = dependencyHandler.create(value)
-        }
-
-    private var _dependencyArtifact: Dependency? = null
-    internal val dependencyArtifact: Dependency get() {
-        val artifact = _dependencyArtifact
-        requireNotNull(artifact) { "Reporter dependency is not set!" }
-        return artifact
+    val reporterId: String = name,
+    var fileExtension: String = reporterId,
+    @Transient var dependency: Any? = null,
+) : Serializable {
+    companion object {
+        private const val serialVersionUID: Long = 2012775L
```

Comment:
Why is this now serializable?

Reply:
To object needs to implement `Serializable` interface to be able to pass as input parameter to the worker.